### PR TITLE
[7] chore: clean up TODOs after refactoring

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

Figma: https://www.figma.com/file/66BWGSBbkqLYCig8gLKj29/Fee-0---Modal?type=design&node-id=290-275&mode=design&t=C0K7pAWDEUFqGayJ-0

PRs list: https://github.com/cowprotocol/cowswap/labels/Trade%20confirm

In this PR I will add fixes after testing and code review. This is the final PR in the refactoring.

  # To Test

Actually the whole app regression test.
The trade flow UI is changed, now all modals must be displayed as screens + UI changes from Figma.
TODO: fix e2e tests

### CODE REVIEW

**Review by anxolin — COMMENTED** [review]
The UI looks way better!!
I really like the new UX.


@fairlighteth, could we try to add an UI effect when the counter arrives to zero? Like a simmering or lightning effect? It goes from 1 to 10 all the sudden. 

Alternatively, or in addition, we could use a donut like:

https://github.com/cowprotocol/cowswap/assets/2352112/acbaff08-3586-4ae0-a715-f7aa7a169e77



See
https://github.com/cowprotocol/cowswap/assets/2352112/c9e00a85-8e64-4589-8681-eaf053d4c9fd



@shoom3301 I think we need to close the confirmation modal if we change the network, look at what happens:
Screenshot at Feb 01 11-57-04

**Comment by alfetopito** [comment]
I thought we were getting rid of this modal:
image

Not yet?

**Comment by alfetopito** [comment]
There's a quote refresh on limit orders 
image

I believe we agreed to not have it there?

**Comment by alfetopito** [comment]
I have unlimited approval for COW, but it shows this number instead
image

Approval tx https://sepolia.etherscan.io/tx/0xa331c756f98119603dbfe6235091836f0798c60f7ec37de7e1a14fa0b61bc746

**Review by alfetopito — COMMENTED** [review]
Overall it's working great!

Left a few comments on the behavior, but nothing major AFAICT

**Comment by shoom3301** [comment]
>think we need to close the confirmation modal if we change the network, look at what happens:

@anxolin thanks! Fixed: https://github.com/cowprotocol/cowswap/pull/3756/commits/f19b740194880202ff210a5d919364665d4fb711

**Comment by shoom3301** [comment]
>I thought we were getting rid of this modal:

@alfetopito sorry, I'm not aware about this. Could you give more info please? When and why we decided to remove the modal?

**Comment by shoom3301** [comment]
>There's a quote refresh on limit orders

@alfetopito thanks! Removed it: https://github.com/cowprotocol/cowswap/pull/3756/commits/82ab9bda3bea0497c597b86715195bbab9ae953b

**Comment by shoom3301** [comment]
>I have unlimited approval for COW, but it shows this number instead

@alfetopito good catch! Fixed: https://github.com/cowprotocol/cowswap/pull/3756/commits/e8227403e89abc4bc72442bbe5907b373c8f034c

**Comment by elena-zh** [comment]
Hey @shoom3301 , great job!
Some my notes: 
1. There is no 'error' modal when: call a sign transaction (the event is sent to the wallet), then press on the back arrow in the modal, then open MM and reject signing an order
no _ missalignment
2. Error modal header looks like missaligned a bit
3.  There is no 'error' modal when reject a Limit order signing (with/without pressing on  he back button on the confirm modal)
image
4. No error message when reject ETH-flow order signing in Expert mode and in case I described in p.1
5. I see ETH token's approved qty in this PR
image
6. No error message when reject a token approval TX
7. For the Convert TX, should we show the back arrow on the waiting modal? 
image
8. Same question for tokens approvals from the Tokens page and Cancel order modals
image
9. When a wallet is not connected, Approve button is in the loading state. It should be displayed and should call Connect to the wallet modal
image
10. I don't see different types of warnings on the confirm modal: fees amount
image
11. Custom recipient warning (swap) + there is no recipient field on the modal: 
image
image
12. Not possible to calculate price impact - swap
13. When I start signing the order, and close 'price impact warning', I get this 
image
image
14. Mobile view: modals do not look good in iOS
image
image
15. It would be great to scroll to the top in a mobile view when placing a limit order: the Review button is at the bottom, the confirm modal appears at the top
16. ETH-flow process: looks like circles are missaligned a bit
image

17. No 'pending orders' pop-up when close the progress bar modal 
image
18. (Leandro reported this as well): fully approved token shows an amount 
image
19. I've got a crash when I switched from Tokens page in Sepolia to the Mainnet
image
20. No error when I reject wrapping for the classic eth-flow 
image
21. no waiting for confirm modal when place an order in Expert mode (we have it on Prod)
22. Approve action for the classic ETH-flow is broken 
image
23. Safe+bundling: the button's name on the confirm modal should be 'Confirm (Approve WETH and Swap)
image
image

24. Is this OK to see 'Unwrapping' modal outside the widget in Safe? 
image

Thanks!

File: apps/cowswap-frontend/src/modules/trade/containers/TradeConfirmModal/index.tsx
Comment (alfetopito) [inline apps/cowswap-frontend/src/modules/trade/containers/TradeConfirmModal/index.tsx:94]: Nitpick:

```suggestion
    return <TransactionErrorContent isScreenMode message={error} onDismiss={onDismiss} />
```

**Comment by alfetopito** [comment]
> > I thought we were getting rid of this modal:
> 
> @alfetopito sorry, I'm not aware about this. Could you give more info please? When and why we decided to remove the modal?

I understood the point of the new confirmation styles was to keep them visible while the user signed the order, so they could check the amounts being signed.
Thus, this intermediary sign modal would be removed.

**Comment by shoom3301** [comment]
@elena-zh thanks a lot for the comments!

1. Not a new issue, works the same on prod
2. Fixed
3. Not a new issue, works the same on prod
4. Not a new issue, works the same on prod
5. Fixed
6. Fixed
7. Fixed
8. Fixed
9. Fixed
10. Fixed
11. Fixed
12. Fixed
13. Fixed (p.s. it's reproducible on prod as well)
14. Fixed
15. Fixed
16. @fairlighteth could you help with it please?
17. Work in progress
18. Fixed
19. Fixed
20. Not a new issue, works the same on prod
21. Work in progress
22. Couldn't reproduce
23. Fixed
24. Couldn't reproduce

**Comment by shoom3301** [comment]
image

image

image

image

image

@fairlighteth @elena-zh I also refactored classic eth-flow

**Comment by shoom3301** [comment]
@elena-zh could you please recheck zero-approve flow? (USDT)

File: apps/cowswap-frontend/src/common/pure/NewModal/index.tsx
Comment (alfetopito) [inline apps/cowswap-frontend/src/common/pure/NewModal/index.tsx:154]: Use `modalMode` to match the prop from the other stuff?

File: apps/cowswap-frontend/src/common/pure/TransactionErrorContent/index.tsx
Comment (alfetopito) [inline apps/cowswap-frontend/src/common/pure/TransactionErrorContent/index.tsx:51]: Same here. I'd prefer to have a standard name for this.

**Comment by elena-zh** [comment]
Hey @shoom3301 , great job, thank you!
Some issues that I still face: 
1. Error signing order is not detected on Limit orders page (placing modal hangs)
limit order reject
2. I see 'back' arrow on the token approval modal on the Tokens page
image
3. When a wallet is not connected, you hidden the Approve button. But it exists on Prod and calls the Connect to wallet modal :)
approve
4. Custom recipient on Swap form: there is still no warning. See the example on the limit orders: 
image
5. Custom recipient warning on Limit orders: there is no need in 'Output will be send' message
image
Overall, it would be great to add a Recipient field to the modal on the Swap page and remove this message from there as well :)
6. I have reported #3788 issue for @fairlighteth to fix 16 point
7. approved token shows an amount of a token issue is not fixed

image
8. Limit orders: call cancel order modal, press on the Back arrow. Reject signing cancellation request in MM. Call cancel order modal once again --> you will get 'Error signing order' modal
cancel

Thanks!

File: apps/cowswap-frontend/src/modules/swap/containers/ConfirmSwapModalSetup/index.tsx
Comment (alfetopito) [inline apps/cowswap-frontend/src/modules/swap/containers/ConfirmSwapModalSetup/index.tsx:76]: Nitpick

```suggestion
          {!priceImpact.priceImpact && <NoImpactWarning isAccepted withoutAccepting />}
```

File: apps/cowswap-frontend/src/modules/usdAmount/hooks/useUsdPrice.ts
Comment (alfetopito) [inline apps/cowswap-frontend/src/modules/usdAmount/hooks/useUsdPrice.ts:37]: How about:

```suggestion
  if (!price || price.currency.chainId !== currency?.chainId) return null

  return price
```

Comment (alfetopito) [reply apps/cowswap-frontend/src/modules/usdAmount/hooks/useUsdPrice.ts:37]: If we want to be extra compact:

```suggestion
  return !price || price.currency.chainId !== currency?.chainId ? null : price
```

But that might be less readable...

**Review by alfetopito — COMMENTED** [review]
A few nitpicks in the code, nothing major.

**Comment by elena-zh** [comment]
Hey @shoom3301 , one more weird case I've faced: 
1. Start placing an order with a permittable token (not approved)
2. When there is a request to sign a TX (approve) in the wallet, press back on the Approve modal
image

Placing order modal will appear (regular one). And it will not be possible to close it by pressing on the back button once again
back


Same happens when press on the Back arrow from the 2nd screen (placing order after approval signing). And when I reject an order signing in the wallet. 

-----------------

Also, I have noticed that on the Swap form it says limit (for a permittable tokens flow): 
image

**Comment by elena-zh** [comment]
One more proposal: when place a swap order, there is a button at the bottom that navigates to the Game page. Maybe we should add an additional one ('Continue', like on the Limit orders) that will close the modal? I know that there is a 'back' arrow that closes it, but it is easier to press on the big button instead of the arrow :)
continue

@fairlighteth , WDYT?

**Comment by shoom3301** [comment]
@elena-zh thank you!
Fixed both issues

File: apps/cowswap-frontend/src/common/pure/NewModal/index.tsx
Comment (shoom3301) [reply apps/cowswap-frontend/src/common/pure/NewModal/index.tsx]: Fixed: https://github.com/cowprotocol/cowswap/pull/3756/commits/08afda61eca55c270ff0dac10aa8a68c5a88ef6d

File: apps/cowswap-frontend/src/common/pure/TransactionErrorContent/index.tsx
Comment (shoom3301) [reply apps/cowswap-frontend/src/common/pure/TransactionErrorContent/index.tsx]: Fixed: https://github.com/cowprotocol/cowswap/pull/3756/commits/08afda61eca55c270ff0dac10aa8a68c5a88ef6d

File: apps/cowswap-frontend/src/modules/swap/containers/ConfirmSwapModalSetup/index.tsx
Comment (shoom3301) [reply apps/cowswap-frontend/src/modules/swap/containers/ConfirmSwapModalSetup/index.tsx]: Fixed: https://github.com/cowprotocol/cowswap/pull/3756/commits/08afda61eca55c270ff0dac10aa8a68c5a88ef6d

File: apps/cowswap-frontend/src/modules/usdAmount/hooks/useUsdPrice.ts
Comment (shoom3301) [reply apps/cowswap-frontend/src/modules/usdAmount/hooks/useUsdPrice.ts]: Fixed: https://github.com/cowprotocol/cowswap/pull/3756/commits/08afda61eca55c270ff0dac10aa8a68c5a88ef6d

**Comment by shoom3301** [comment]
@elena-zh also fixed issues from https://github.com/cowprotocol/cowswap/pull/3756#issuecomment-1932141052,
besides https://github.com/cowprotocol/cowswap/issues/3788

File: apps/cowswap-frontend/src/common/pure/NewModal/index.tsx
Comment (shoom3301) [reply apps/cowswap-frontend/src/common/pure/NewModal/index.tsx]: Nice catch! Thanks!
Fixed

**Review by elena-zh — APPROVED** [review]
Looks good now!
I have created #3833  issue with some enhancements related to the Recipient value on the modal, but it can be addressed later.

**Pull request merged**
